openapi.yaml

```diff
@@ -1,279 +1,662 @@
-swagger: '2.0'
+openapi: '3.0.2'
```

Comment:
using 3.0.2 instead of 3.0.3 because my vscode extensions don't support 3.0.3. If that's an issue I can work around the extensions problem, but I don't think the two are very different.